### PR TITLE
Add release notes for version 2.0.0-rc1

### PR DESCRIPTION
**Pull request by lezzago**
Signed-off-by: Ashish Agrawal <ashisagr@amazon.com>

### Description
* Add release notes for version 2.0.0-rc1
 
### Issues Resolved
N/A
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
Should the notes include https://github.com/opensearch-project/alerting-dashboards-plugin/pull/226?

File: release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.0.0-rc1.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+## Version 2.0.0.0-rc1 2022-04-25
+Compatible with OpenSearch Dashboards 2.0.0-rc1
+
+### Enhancements 
+* Implemented UX support for configuring doc level monitors. ([#218](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/218))
+* Integrate Alerting Dashboards with Notifications Plugin ([#220](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/220))
+* Remove disabled buttons and update Destination flows to reflect read-only state ([#221](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/221))
```
Comment (getsaurabh02) [inline release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.0.0-rc1.md:7]: wondering if this should go as part of ``Refactor``?

**Comment by getsaurabh02** [comment]
@AWSHurneyt Isn't #226 already covered as part of ([#218 - Implemented UX support for configuring doc level monitors.](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/218)) since this is first time doc level alerting is being released?

File: release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.0.0-rc1.md (hunk 1 above)
Comment (getsaurabh02) [reply release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.0.0-rc1.md:7]: I meant for #221 only

**Comment by lezzago** [comment]
> 

I agree with @AWSHurneyt since based on existing standard, we would still want to include #226 under bug fixes

File: release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.0.0-rc1.md (hunk 1 above)
Comment (lezzago) [reply release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.0.0.0-rc1.md]: done

**Review by AWSHurneyt — APPROVED** [review]

**Review by qreshi — APPROVED** [review]

**Pull request merged**